### PR TITLE
Choropleth Tool Improvements

### PR DESCRIPTION
**Pull request by chejennifer**
- Few simple changes to make title a little more readable: remove underscore, few replacements in the words in the title. More efforts will be needed to make title readable
- Show the selected stat var in the menu when loading the choropleth tool for the first time
- keep the stat var menu open and keep the current option selected when navigating to different place (ie. when clicking on a region in the choropleth or when clicking a breadcrumb)

test (14)

### CODE REVIEW

File: static/js/tools/choropleth/choropleth.tsx [hunk 1]
```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
...
+  /**
+   * Redirects the webclient to a particular geo. Handles breadcrumbs in redirect.
+   * @param {string} geoDcid to redirect to.
+   * @param {string} curGeo human-readable current geo, e.g. United States when geoDcid
+   *                 is country/USA.
+   */
+  private redirectToGeo = (geoDcid: string, curGeo: string): void => {
+    const url = new URL(window.location.href);
+    const currGeoId = url.searchParams.get("geoDcid");
+    let baseUrl = "/tools/choropleth";
```
Comment (shifucun) [inline static/js/tools/choropleth/choropleth.tsx:620]: js string template would be better:

`/tools/choropleth${buildChoroplethParams()}&geoDcid=${geoDcid}&bc=`

File: static/js/tools/choropleth/choropleth.tsx [hunk 2]
```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
...
+            </span>
+          );
+        }
+      });
+      breadcrumbsDisplay.push(<span key={curGeo}>{curGeo}</span>);
+    }
+    // Add breadcrumbs + current geoId
+    // Example: "USA" > "FL" > "Miami-Dade"
+    // Where Miami-Dade is the current id
+    ReactDOM.render(breadcrumbsDisplay, document.getElementById("breadcrumbs"));
```
Comment (shifucun) [inline static/js/tools/choropleth/choropleth.tsx:676]: Might not be good to call ReactDOM.render in a react Component. How about set the state of a children component for the bradcrumbs?

File: static/js/tools/choropleth/choropleth.tsx (hunk 1 above)
Comment (beets) [reply static/js/tools/choropleth/choropleth.tsx:620]: you could also use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

File: static/js/tools/choropleth/choropleth.tsx [hunk 3]
```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
...
+      breadCrumbsList.push({ geoId: currGeoId, geoName: curGeo });
+    } else {
+      breadCrumbsList = breadCrumbsList.slice(0, idxOfCurr);
+    }
+    this.setState({
+      breadCrumbs: breadCrumbsList,
+    });
+    for (let i = 0; i < breadCrumbsList.length; i++) {
+      baseUrl += i === 0 ? "" : ";";
+      baseUrl += breadCrumbsList[i].geoId + "~" + breadCrumbsList[i].geoName;
```
Comment (beets) [inline static/js/tools/choropleth/choropleth.tsx:639]: would be good to use a constant for the separator (helps with readability and maintainability too)

File: static/js/tools/choropleth/choropleth.tsx [hunk 4]
```diff
@@ -576,6 +608,74 @@ class ChoroplethMap extends Component<PropsType, StateType> {
...
+    history.pushState({}, null, baseUrl);
+    this.loadGeoJson();
+  };
+
+  /**
+   * Generates the breadcrumbs text from browser url.
+   * @param {string} curGeo human-readable current geo to display
+   * at end of list of hierarchy of locations.
+   */
+  private generateBreadCrumbs = (curGeo: string): void => {
```
Comment (beets) [inline static/js/tools/choropleth/choropleth.tsx:650]: Ideally, we should turn the breadcrumbs into a component. But we can add it as a todo for now

**Review by beets — COMMENTED** [review]
There's some logic from the timelines tool for the stat var title that we should try to bring in. Could you add a TODO for that?

File: static/js/tools/choropleth/choropleth.tsx [hunk 5]
```diff
@@ -615,12 +617,14 @@ class ChoroplethMap extends Component<PropsType, StateType> {
...
-    baseUrl += "&geoDcid=" + geoDcid;
-    baseUrl += "&bc=";
+    const currGeoId = new URLSearchParams(window.location.search).get(
+      "geoDcid"
+    );
+    let baseUrl = `/tools/choropleth${buildChoroplethParams([
+      "statVar",
+      "pc",
+      "mdom",
+    ])}&&geoDcid=${geoDcid}&bc=`;
```
Comment (beets) [inline static/js/tools/choropleth/choropleth.tsx:627]: double ampersands? also looks like you could pass in geoDcid into the list of args for buildChoroplethParams to preserve.

Comment (chejennifer) [reply static/js/tools/choropleth/choropleth.tsx:627]: I will fix the double ampersands. buildChoroplethParams will set values to what is currently in the url, but we want the geoDcid for the next geo we're redirecting to

Comment (beets) [reply static/js/tools/choropleth/choropleth.tsx]: Ah yes, thanks for catching that. Readability nit for future: the code should be localized with where it's used (i.e. get currGeoId right before using it).

I'll approve and let's submit.

**Review by beets — APPROVED** [review]

**Pull request merged**